### ISSUE TITLE
Freezing when launching external viewer on Linux

### ISSUE THREAD

**Issue by Dansoftowner** [Document-Archiver/com.sophisticatedapps.archiving.document-archiver#13]
Hi @stephansann ,

I tried Document Archiver on my Ubuntu machine.
I experienced that the app freezes when I try to open the external editor for the particular file.

![Screenshot](https://user-images.githubusercontent.com/49782833/117342317-6312a280-aea3-11eb-895b-0eaf2efcf415.png)

The funny thing is that no exception occurs.

Fortunately, I had the same problem with the `java.awt.Desktop` API in my application on Linux systems and now I know what is the solution.
So basically you should execute the `Desktop.getDesktop().open(...)` command on an another `thread`.
I used an ExecutorService:
```java
ExecutorService executor = Executors.newCachedThreadPool(); // just an example
executor.submit(() -> Desktop.getDesktop().open(file));
```

Hope you find this report useful!
Best regards
Daniel

**Comment by stephansann** [comment]
Hi @Dansoftowner 

Thanks a lot for the report.

Could you maybe share the file this is happening with?

Anyhow I will have a look at the code and change it as suggested.

Best regards
Stephan

**Comment by Dansoftowner** [comment]
It happens with every file I tried, the problem is not with the files.
It's basically a bug on linux, but somehow invoking the Desktop API method on another `thread` solves the problem.

**Comment by stephansann** [comment]
Hi @Dansoftowner 

Seems to be okay now in the latest SNAPSHOT-build:
https://repository.sophisticatedapps.com/snapshots/com/sophisticatedapps/archiving/document-archiver/2.1.0-SNAPSHOT/document-archiver-2.1.0-20210510.153821-2-linux.zip

Could you please have a look if okay for you, too?

Thanks
Stephan

**Comment by Dansoftowner** [comment]
Hi @stephansann ,

Now it works fine! :)

Daniel

**Comment by stephansann** [comment]
Hi @Dansoftowner 

Nice - happy to learn it works for you, too.

Thanks again for reporting and also for the suggested solution!

BR, Stephan
